setup.py

```diff
@@ -19,13 +19,23 @@
 import codecs
 import os
 import re
+import setuptools
 from setuptools import setup, find_packages
+from pkg_resources import parse_version
 
 here = os.path.abspath(os.path.dirname(__file__))
 
 with open("{}/README.md".format(here), "r") as fh:
     long_description = fh.read()
 
+setuptools_version = parse_version(setuptools.__version__).base_version
+if setuptools_version < '43.0.0':
+    raise RuntimeError(
+        "To properly install NNCF, please install setuptools>=43.0.0, "
+        "while current setuptools version is {curr}".format(
+        curr=setuptools.__version__
+    ))
```

Comment:
Can we limit this to only the cases that currently fail on Python 3.6? I.e. check that Python is < 3.7 and that the tensorflow installation is requested. Otherwise we will fail in more cases than we should.

Reply:
Or does it really only depend on setuptools and not on Python (e.g. if, for example, the 3.7 distribution includes newer setuptools by default)? Even then we probably still should only fail for TF installation requests.

Reply:
This is an issue unrelated to the python version. On python3.8 and `setuptools<43.0.0` I get the same error.